std-bits/tableau/src/main/java/org/enso/tableau/HyperReader.java

```diff
@@ -0,0 +1,209 @@
+package org.enso.tableau;
+
+import com.tableau.hyperapi.Catalog;
+import com.tableau.hyperapi.Connection;
+import com.tableau.hyperapi.HyperProcess;
+import com.tableau.hyperapi.SchemaName;
+import com.tableau.hyperapi.TableName;
+import com.tableau.hyperapi.Telemetry;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.channels.Channels;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.IntStream;
+import org.enso.table.data.table.Column;
+import org.enso.table.problems.ProblemAggregator;
+
+// ** Class responsible for reading from Tableau Hyper files. */
+public class HyperReader {
+  public static final Path HYPER_PATH = Path.of(getHyperPath());
+  private static HyperProcess process;
+
+  private static final Logger LOGGER = Logger.getLogger("enso-hyper-reader");
+
+  private static String getHyperPath() {
+    if (System.getenv("HYPER_PATH") != null) {
+      return System.getenv("HYPER_PATH");
+    }
+    if (System.getenv("ENSO_DATA_DIRECTORY") != null) {
+      return System.getenv("ENSO_DATA_DIRECTORY") + "/hyper";
+    } else {
+      return switch (OSPlatform.CurrentPlatform) {
+        case WINDOWS -> System.getenv("LocalAppData") + "/enso/hyper";
+        case MAC_ARM64, MAX_X64 -> System.getProperty("user.home")
+            + "/Library/Application Support/org.enso/hyper";
+        case LINUX, OTHER -> System.getProperty("user.home") + "/.local/share/enso/hyper";
+      };
+    }
+  }
```

Comment:
Why `ENSO_DATA_DIRECTORY` and not `ENSO_CACHE_DIRECTORY`?

Also we are duplicating (a minimal portion of) logic from `DistributionManager` - for now probably OK but the more we do it I'm wondering if we should refactor the most basic parts of it to make them re-usable in libs like this.